src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/OrganisationServiceTest.java

```diff
@@ -0,0 +1,378 @@
+        verify(organisationRefreshQueueRepository, times(2))
+                .upsertToOrganisationRefreshQueue(any(), any(), any());
+        verify(profileRefreshQueueRepository, times(1))
+                .setActiveFalse(any(), any());
+    }
+
+    @Test
+    void findAndInsertStaleOrganisationsIntoRefreshQueue_NoActiveOrganisationsTest() {
+        when(profileRefreshQueueRepository.getActiveProfileEntities()).thenReturn(Collections.emptyList());
+
+        organisationService.findAndInsertStaleOrganisationsIntoRefreshQueue();
+
+        verify(profileRefreshQueueRepository, times(1))
+                .getActiveProfileEntities();
+        verify(organisationRefreshQueueRepository, times(0))
+                .upsertToOrganisationRefreshQueue(any(), any(), any());
+        verify(profileRefreshQueueRepository, times(0))
+                .setActiveFalse(any(), any());
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    void findOrganisationChangesAndInsertIntoOrganisationRefreshQueue() {
+        DatabaseDateTime databaseDateTime = mock(DatabaseDateTime.class);
+        when(databaseDateTime.getDate()).thenReturn(mock(Instant.class));
+        when(databaseDateTimeRepository.getCurrentTimeStamp()).thenReturn(databaseDateTime);
+        List<AccessTypesEntity> allAccessTypes = new ArrayList<>();
+        allAccessTypes.add(new AccessTypesEntity(1L, "some json"));
+        when(accessTypesRepository.findAll()).thenReturn(allAccessTypes);
+
+        List<BatchLastRunTimestampEntity> allBatches = new ArrayList<>();
+        allBatches.add(new BatchLastRunTimestampEntity(1L, LocalDateTime.of(2023, 12, 30, 0, 0, 0, 0),
+                LocalDateTime.of(2023, 12, 31, 12, 34, 56, 789)));
+        when(batchLastRunTimestampRepository.findAll()).thenReturn(allBatches);
+        List<OrganisationInfo> allOrgs1 = new ArrayList<>();
+        OrganisationInfo organisationInfo1 = buildOrganisationInfo(1);
+        OrganisationInfo organisationInfo2 = buildOrganisationInfo(2);
+        allOrgs1.add(organisationInfo1);
+        allOrgs1.add(organisationInfo2);
+
+        List<OrganisationInfo> allOrgs2 = new ArrayList<>();
+        OrganisationInfo organisationInfo3 = buildOrganisationInfo(3);
+        allOrgs2.add(organisationInfo3);
+
+        ResponseEntity<OrganisationsResponse> organisationsResponse1 = ResponseEntity
+                .ok(OrganisationsResponse.builder().organisations(allOrgs1).moreAvailable(true).build());
+        ResponseEntity<OrganisationsResponse> organisationsResponse2 = ResponseEntity
+                .ok(OrganisationsResponse.builder().organisations(allOrgs2).moreAvailable(false).build());
+
+        when(prdService.retrieveOrganisations(anyString(), anyInt(), anyInt()))
+                .thenReturn(organisationsResponse1, organisationsResponse2);
+
+        organisationService.findOrganisationChangesAndInsertIntoOrganisationRefreshQueue();
+
+        verify(organisationRefreshQueueRepository, times(2))
```

Comment:
Would be nice to have a capture verifying updated data passed to queue repository calls on at least one of the happy path tests in the unit test or as an assert on the db values after the equivilant intergation tests.